### PR TITLE
Scaffold new dev docs

### PR DESCRIPTION
**Pull request by mltejera**
This PR is scaffolding our new documentation directory in the repo.

I will fast follow with individual PR's for each file, with the assigned reviewer giving me feedback.

**Related issues:**

Related RFC #30274
Related Epic #30642

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/Avatar-mount_base.html">643</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/Avatar-mount.html">615</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/Button-mount_base.html">290</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/Button-mount.html">308</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/Field-mount_base.html">1149</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/Field-mount.html">1117</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/FluentProvider-mount_base.html">700</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/FluentProvider-mount.html">705</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/FluentProviderWithTheme-mount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/FluentProviderWithTheme-mount.html">91</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">59</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">65</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">70</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/MakeStyles-mount_base.html">867</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/MakeStyles-mount.html">866</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/Persona-mount_base.html">1749</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/Persona-mount.html">1704</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/SpinButton-mount_base.html">1381</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30483/perf-test-react-components/SpinButton-mount.html">1412</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Review by khmakoto — APPROVED** [review]

File: CONTRIBUTING.md
Comment (JustSlone) [inline CONTRIBUTING.md:1]: Can we wait to add contributing.md until it has content? This brings the repro into a strange state where we have a contributing.md (which will cause certain checks to pass), but yet it's empty containing no content.

**Comment by JustSlone** [comment]
Thanks for spinning this up!

On the `__documentation__` folder name. I think should avoid the double `__`'s unless there is a strong reason to use this. It's odd, I assume this is to affect the sort order of the folder. To either hide it or make it stand out more. Perhaps there is something I'm missing here but it seems like a strange practice that will draw unnecessary attention within the repo. 

This treatment, either makes the folder really obvious, or makes it hard to find given it's now above the dotfile folders which are ignored. I think the outcome here will vary depending on the engineer with some seeing it and some ignoring it, I don't think we want that outcome. 
image

I think it would be better to just have the folder be `docs` or `documentation` (I prefer `docs` following the pattern of shorter folder names where reasonable). Let's just let it fit within the folders there are already a small number, and there will be one less with `rfcs` moving under `docs`.

**Comment by JustSlone** [comment]
Every unowned file is owned by `fluentui-admins`
https://github.com/microsoft/fluentui/blob/94468f9267b77b44cbc9114c9eee6957496c2999/.github/CODEOWNERS#L20

I'd suggest adding a code owner entry for the `docs` folder, following the pattern of `rfcs`
https://github.com/microsoft/fluentui/blob/94468f9267b77b44cbc9114c9eee6957496c2999/.github/CODEOWNERS#L328

File: CONTRIBUTING.md
Comment (mltejera) [reply CONTRIBUTING.md:1]: Agreed - I realized there is also an existing 'contributing' markdown with existing content that this unnesesarily competes with.

File: docs/react-v9/contributing/dev-env.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Todo: Describe the set up process from start to getting the demo app running locally.
+// Assignees: Mason
+// Reviewer:
```
Comment (mltejera) [inline docs/react-v9/contributing/dev-env.md:3]: Mak volunteered to review this one.

File: docs/react-v9/contributing/dev-workflow.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Todo: Describe the process for creating a branch, doing work, committing, preparation for PR's, review process, and getting it checked in.
+// Assignees:
```
Comment (mltejera) [inline docs/react-v9/contributing/dev-workflow.md:2]: Mason to author, waiting on volunteer for reviewer.

File: docs/react-v9/contributing/command-cheat-sheet.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Todo: Simple table of common commands used in the repo.
+// Assignees:
```
Comment (mltejera) [inline docs/react-v9/contributing/command-cheat-sheet.md:2]: Mason to author, waiting for reviewer

**Review by marcosmoura — APPROVED** [review]

File: CONTRIBUTING.md
Comment (JustSlone) [inline CONTRIBUTING.md:1]: It looks like this file got added back? We should remove it from this PR

**Comment by JustSlone** [comment]
Is this PR ready to merge, or will there be work to complete the //TODO's? (If you could elaborate on the plan and details for the PR in the description that will help)

I'm a little unsure about adding actual files with //TODO and assignees, should that be issues instead? 

If we have high confidence that we'll fast follow to fix these it's fine, but I'd hate for us to have empty md files with //TODO in them in the repo for an extended period of time (where as having an issue we haven't gotten around to yet is fine). 

Thanks!

**Review by tudorpopams — APPROVED** [review]

**Review by JustSlone — APPROVED** [review]
Thanks for the updates, PR looks good!

**Review by layershifter — APPROVED** [review]

**Pull request merged**